### DISCUSSION TITLE
Some feature ideas

### DISCUSSION THREAD

**Discussion by dracarys18** [spacecowboy/Feeder#228] [Ideas]
- We can have a global flag for showing full articles for all of the feeds. It would be in this flow
```kotlin
if (global_flag) {
	if(local_flag) {
		enable_full_article
	}
}
```
- Add a select all button for selecting all feeds while deleting the feed.
- Have an option to export the OPML file to gist.

Happy to work on any of these features if needed

**Comment by spacecowboy** [comment]
> Have an option to export the OPML file to gist.

What's a `gist` in this context?

**Reply by dracarys18** [reply]
https://gist.github.com/

**Reply by spacecowboy** [reply]
Oh. Though it was some kind of file format since you said export. That's definitely out of scope

**Comment by spacecowboy** [comment]
> Add a select all button for selecting all feeds while deleting the feed.

This would be good. 

https://github.com/spacecowboy/Feeder/blob/aced24922dfe5ad66a2d43395a4c7db9bc1d24e4/app/src/main/java/com/nononsenseapps/feeder/ui/compose/deletefeed/DeleteFeedScreen.kt#L39

**Reply by dracarys18** [reply]
Was busy with work, But I have created a PR for this please review

https://github.com/spacecowboy/Feeder/pull/234

**Comment by dracarys18** [comment]
Sure I will create a PR for this

**Reply by spacecowboy** [reply]
Cool.

I think it would work well with a overflow button in the corner (see screenshot).

One thing I've always liked is the ability to invert the selection (see 2nd screenshot). But feel no pressure to implement that.

![Screenshot_20240305_160531_Feeder](https://github.com/spacecowboy/Feeder/assets/223655/ef3bf4e4-ec8e-41c9-bb33-adf2041bd934)
![Screenshot_20240305_160543_Transdroid](https://github.com/spacecowboy/Feeder/assets/223655/8b67f7e2-81ff-4d53-86b4-79079fc25e45)

**Reply by dracarys18** [reply]
Invert selection is deselect right? That can be done

**Reply by spacecowboy** [reply]
no so invert is like "opposite".

say I have 100 feeds and I want to delete 99. Then I'd select the 1 feed I want to keep, then pick "Invert selection". Then all feeds *except* the one I selected originally are now selected.

**Comment by spacecowboy** [comment]
PR in #234
